presto-hive/src/main/java/com/facebook/presto/hive/RetryDriver.java

```diff
@@ -137,13 +139,16 @@ public RetryDriver stopOnIllegalExceptions()
                 return callable.call();
             }
             catch (Exception e) {
+                suppressedExceptions.add(e);
                 e = exceptionMapper.apply(e);
                 for (Class<? extends Exception> clazz : exceptionWhiteList) {
                     if (clazz.isInstance(e)) {
+                        addSuppressed(e, suppressedExceptions);
                         throw e;
                     }
                 }
                 if (attempt >= maxAttempts || Duration.nanosSince(startTime).compareTo(maxRetryTime) >= 0) {
+                    addSuppressed(e, suppressedExceptions);
```

Comment:
I am concerned what happens if `maxAttempts` is large. We can let the `suppressedExceptions` grow large, leading to unreadably-long stacktrace.
Should we have any kind of limit on `suppressedExceptions.size()`?

Reply:
(this is optional)

Reply:
I thought about this too, but didn't bother for now. We can add some limit if this becomes a real problem in the future.